aws/resource_aws_efs_access_point.go

```diff
@@ -0,0 +1,404 @@
+			}
+
+			mt := resp.AccessPoints[0]
+
+			log.Printf("[DEBUG] Current status of %q: %q", *mt.AccessPointId, *mt.LifeCycleState)
+			return mt, *mt.LifeCycleState, nil
+		},
+		Timeout:    10 * time.Minute,
+		Delay:      2 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for EFS access point (%s) to create: %s", d.Id(), err)
+	}
+
+	log.Printf("[DEBUG] EFS access point created: %s", *ap.AccessPointId)
+
+	return resourceAwsEfsAccessPointRead(d, meta)
+}
+
+func resourceAwsEfsAccessPointUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).efsconn
+
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
+
+		if err := keyvaluetags.EfsUpdateTags(conn, d.Id(), o, n); err != nil {
+			return fmt.Errorf("error updating EFS file system (%s) tags: %s", d.Id(), err)
+		}
+	}
+
+	return resourceAwsEfsAccessPointRead(d, meta)
+}
+
+func resourceAwsEfsAccessPointRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).efsconn
+	resp, err := conn.DescribeAccessPoints(&efs.DescribeAccessPointsInput{
+		AccessPointId: aws.String(d.Id()),
+	})
+	if err != nil {
+		if isAWSErr(err, efs.ErrCodeAccessPointNotFound, "") {
+			log.Printf("[WARN] EFS access point %q could not be found.", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading EFS access point %s: %s", d.Id(), err)
+	}
+
+	if hasEmptyAccessPoints(resp) {
+		return fmt.Errorf("EFS access point %q could not be found.", d.Id())
+	}
+
+	ap := resp.AccessPoints[0]
+
+	log.Printf("[DEBUG] Found EFS access point: %#v", ap)
+
+	d.SetId(*ap.AccessPointId)
+
+	fsARN := arn.ARN{
+		AccountID: meta.(*AWSClient).accountid,
+		Partition: meta.(*AWSClient).partition,
+		Region:    meta.(*AWSClient).region,
+		Resource:  fmt.Sprintf("file-system/%s", aws.StringValue(ap.FileSystemId)),
+		Service:   "elasticfilesystem",
+	}.String()
+
+	d.Set("file_system_arn", fsARN)
+	d.Set("file_system_id", ap.FileSystemId)
+	d.Set("arn", ap.AccessPointArn)
+	d.Set("owner_id", ap.OwnerId)
+
+	if err := d.Set("posix_user", flattenEfsAccessPointPosixUser(ap.PosixUser)); err != nil {
+		return fmt.Errorf("error setting posix user: %s", err)
+	}
+
+	if err := d.Set("root_directory", flattenEfsAccessPointRootDirectory(ap.RootDirectory)); err != nil {
+		return fmt.Errorf("error setting root directory: %s", err)
+	}
+
+	if err := d.Set("tags", keyvaluetags.EfsKeyValueTags(ap.Tags).IgnoreAws().Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsEfsAccessPointDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).efsconn
+
+	log.Printf("[DEBUG] Deleting EFS access point %q", d.Id())
+	_, err := conn.DeleteAccessPoint(&efs.DeleteAccessPointInput{
+		AccessPointId: aws.String(d.Id()),
+	})
+	if err != nil {
+		return err
```

Comment:
Same here regarding error context:

```suggestion
		return fmt.Errorf("error deleting EFS Access Point (%s): %w", d.Id(), err)
```